File: app/src/main/java/org/stepik/android/adaptive/pdd/ui/activity/StudyActivity.java
package org.stepik.android.adaptive.pdd.ui.activity;

import android.support.v4.app.Fragment;
import android.widget.FrameLayout;

import org.stepik.android.adaptive.pdd.R;
import org.stepik.android.adaptive.pdd.ui.fragment.CardsFragment;
import org.stepik.android.adaptive.pdd.util.AchievementManager;

import java.util.concurrent.TimeUnit;

import io.reactivex.Completable;
import io.reactivex.android.schedulers.AndroidSchedulers;

public class StudyActivity extends FragmentActivity {

    @Override
    protected void onResume() {
        super.onResume();

        Completable.fromRunnable(() -> {})
                .delay(2, TimeUnit.SECONDS)
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(() -> {
                    AchievementManager.INSTANCE.show((FrameLayout) findViewById(R.id.fragment_container));
                });
    }

    @Override
    protected Fragment createFragment() {
        return new CardsFragment();
    }
}

File: app/src/main/java/org/stepik/android/adaptive/pdd/ui/activity/StudyActivity.kt
package org.stepik.android.adaptive.pdd.ui.activity

import android.widget.FrameLayout

import org.stepik.android.adaptive.pdd.R
import org.stepik.android.adaptive.pdd.ui.fragment.CardsFragment

import org.stepik.android.adaptive.pdd.core.presenter.contracts.AchievementView
import org.stepik.android.adaptive.pdd.data.model.Achievement
import org.stepik.android.adaptive.pdd.ui.animation.AchievementAnimations
import org.stepik.android.adaptive.pdd.util.AchievementManager

class StudyActivity : FragmentActivity(), AchievementView {
    private var isPlayingAchievementAnimation = false

    override fun showAchievement(achievement: Achievement) {
        isPlayingAchievementAnimation = true
        AchievementAnimations.show(findViewById(R.id.fragment_container) as FrameLayout, achievement).withEndAction {
            isPlayingAchievementAnimation = false
            AchievementManager.notifyQueue()
        }
    }

    override fun canShowAchievement() = !isPlayingAchievementAnimation

    override fun onStart() {
        super.onStart()
        AchievementManager.attachView(this)
    }

    override fun onStop() {
        AchievementManager.detachView(this)
        super.onStop()
    }

    override fun createFragment() = CardsFragment()
}
